packages/js/src/interpreters.ts

```diff
@@ -88,7 +88,9 @@ export const size: Interpret<Field<number>, AnyObject | unknown[]> = (node, obje
     : test(items);
 };
 
-export const regex = testValueOrArray<RegExp, string>((node, value) => node.value.test(value));
+export const regex = testValueOrArray<RegExp, string>(
+  (node, value) => value !== undefined && node.value.test(value)
```

Comment:
I think the best way is to check that value is a string. Otherwise there will be a similar issue with  `null`